File: ValidateCredentials/getVaultCredentials.js

const { DefaultAzureCredential } = require('@azure/identity');
const { SecretClient } = require('@azure/keyvault-secrets');


async function getVaultCredentials(vaultUri, secretName) {
    const credential = new DefaultAzureCredential();
    const secretClient = new SecretClient(vaultUri, credential);

    try {
        const secret = await secretClient.getSecret(secretName);

        // console.log(secret);
        return secret;
    } catch (error) {
        console.error('Error fetching secret from Azure Key Vault:', error);
        throw error;
    }

}



module.exports = { getVaultCredentials };

File: SQL/getVaultCredentials.py
from azure.identity import DefaultAzureCredential
from azure.keyvault.secrets import SecretClient

def getSQLVaultCredentials(vault_uri , secret_name):
    credential = DefaultAzureCredential()

    secretclient =  SecretClient(vault_url=vault_uri, credential=credential)

    secret = secretclient.get_secret(secret_name)

    return secret



secret_val = getSQLVaultCredentials("https://sih-data.vault.azure.net/" , "sihadmin")

print(secret_val.value)

